### ISSUE TITLE
Add ability to pass in a file pointer

### ISSUE THREAD

**Issue by sidchilling** [rmccue/Requests#34]
How to post file data / upload file using requests? I can;t find any documentation on this.

Like in `HttpRequest`, we have a method called `addPostFile()`

**Comment by masiorama** [comment]
Not yet available in the last build. I'm interested in it too.

**Comment by captn3m0** [comment]
PHP 5.5 now has CurlFile support: https://secure.php.net/manual/en/class.curlfile.php

**Comment by fbm-static** [comment]
Anyone have a fork making this work?

**Comment by captn3m0** [comment]
I just shifted to guzzle, which is doing really well in terms of development.

**Comment by fbm-static** [comment]
Yeah, but the main selling point of this library was 5.2 compatibility (not my decision!) which Guzzle doesn't offer.

**Comment by soulseekah** [comment]
Related #289
